executor/table_reader.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/distsql"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/plan"
+	"github.com/pingcap/tidb/statistics"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/ranger"
+	tipb "github.com/pingcap/tipb/go-tipb"
+	"golang.org/x/net/context"
+)
+
```

Comment:
Do we need to move `_ Executor = &TableReaderExecutor{}` here?

Reply:
No need to do so, actually. `baseExecutor` implements all method of `Executor`'s interface. `TableReaderExecutor` embed `baseExecutor` as anonymous 
variable which makes sure `TableReaderExecutor` implementing `Executor` interface.